documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,505 @@
+
+**All interaction with your resource attributes always occur through an action**. In our resource we are using the default actions for `:create, :read, :update, :destroy` along with a custom action `:by_id`.
+
+`:create` and `:update` and `:destroy` actions require a changeset. Ash changesets are conceptually similar to [Ecto changesets](https://hexdocs.pm/ecto/Ecto.Changeset.html). They're data structures which represent an intended change to an Ash resource and provide validation.
+
+The `:read` action takes a query instead of a changeset.
+
+Below is the most verbose way of calling your resource. All other ways of interaction are some kind of shorthand of these. This means at some point a changeset is being created and passed to the API, even if it's encapsulated within another function.
+
+```elixir
+# create post
+new_post =
+  MyAshPhoenixApp.Blog.Post
+  |> Ash.Changeset.for_create(:create, %{title: "hello world"})
+  |>  MyAshPhoenixApp.Blog.create!()
+
+# read all posts
+MyAshPhoenixApp.Blog.Post
+|> Ash.Query.for_read(:read)
+|> MyAshPhoenixApp.Blog.read!()
+
+# get single post by id
+MyAshPhoenixApp.Blog.Post
+|> Ash.Query.for_read(:by_id, %{id: first_post.id})
+|> MyAshPhoenixApp.Blog.read_one!()
+
+# update post
+updated_post =
+  new_post
+  |> Ash.Changeset.for_update(:update, %{content: "hello to you too!"})
+  |> MyAshPhoenixApp.Blog.update!()
+
+# delete post
+first_post
+|> Ash.Changeset.for_destroy(:destroy)
+|> MyAshPhoenixApp.Blog.destroy!()
+```
+
+As I said, this is verbose so Ash has a built in shortcut - The `code_interface`. You may notice this has already been done in your `Post` resource. Here it is again with more explanation:
+
+```elixir
+ code_interface do
+    # defines the API this resource should be called from
+    define_for MyAshPhoenixApp.Blog
+    # defining function Post.create/2 it calls the :create action
+    define :create, action: :create
+    # defining function Post.read_all/2 it calls the :read action
+    define :read_all, action: :read
+    # defining function Post.update/2 it calls the :update action
+    define :update, action: :update
+    # defining function Post.destroy/2 it calls the :destroy action
+    define :destroy, action: :destroy
+    # defining function Post.get_by_id/2
+    # it calls the :by_id action with the argument :id
+    define :get_by_id, args: [:id], action: :by_id
+  end
+```
+
+> Note: The function name doesn't have to match the action name in any way. You could also write:
+>
+> ```elixir
+> define :make, action: :create
+> ```
+>
+> That's perfectly valid and could be called via `Blog.make/2`.
+
+Now we can call our resource like so:
+
+```elixir
+# create post
+new_post = MyAshPhoenixApp.Blog.Post.create!(%{title: "hello world"})
+
+# read post
+MyAshPhoenixApp.Blog.Post.read_all!()
+
+# get post by id
+MyAshPhoenixApp.Blog.Post.get_by_id!(new_post.id)
+
+# update post
+updated_post = MyAshPhoenixApp.Blog.Post.update!(%{content: "hello to you too!"})
```

Comment:
this is missing the first argument of the post to be updated

Reply:
[8a2b35f](https://github.com/ash-project/ash_phoenix/pull/70/commits/8a2b35f9bac7edef885ce5a6f3337fa2d68dfa38)